t/stream-concurrency.c

```diff
@@ -94,7 +94,6 @@ void test_stream_concurrency(void)
     /* we would have free room now that RST of the server-sent side is ACKed */
     ok(server_streambuf->is_detached);
     ok(!client_streams[i]->streams_blocked);
-    ++i;
```

Comment:
Doesn't removing this like cause `client_streams[i]` that already contains an reference to an open stream to be overwritten when `quicly_open_stream` is called below?

I'm not sure what the issue is, but I think it needs to be fixed in a different way.

Reply:
Hi @kazuho, the issue was flagged by AddressSanitizer against the unit test. The issue with the unit test is that there are
eleven memory elements allocated (at line 30), addressable via client_streams[0] to client_streams[10]. First, i = 10 at t/stream-concurrency.c:65. Then, i = 11 at line 100, due to the ++i at line 97. This means that when we get to line 99, client_streams+i in the quicly_open_stream function call refers to client_streams[11], which is the twelfth element, and which can’t be accessed, since only eleven memory elements were defined at line 30.

Initially, when I looked at the intent of the unit test, I looked at the comment "we would have free room now that RST of the server-sent side is ACKed" and interpreted it to mean that the last client_streams connection is no longer being used, so this is why I thought of removing that ++i. However, we could certainly fix the issue by defining quicly_stream_t *client_streams[limit + 2] in line 30, if the intent was to have a test to create one more new client_stream.

Reply:
@sescandor Thank you for the explanation.

I think you are correct that this change would not cause any harm, considering how `client_streams[i]` is used.

That said, could you please instead expand the size of `client_streams[]` by one to avoid the issue? The coding pattern we have here is to use a new slot of the `client_streams` array each time we call `quicly_open_stream`, and see how the states of those streams being opened change as things go.